addon/helpers/target-value.js

```diff
@@ -0,0 +1,20 @@
+import { helper } from '@ember/component/helper';
+import { assert } from '@ember/debug';
+
+export function targetValue([handler]) {
+  assert(
+    `Expected '${handler}' to be a function, if present.`,
+    !handler || typeof handler === 'function'
+  );
+
+  return function (event) {
+    assert(
+      `Expected '${event}' to have a 'target' with a 'value' on it.`,
+      event && event.target && event.target.value
```

Comment:
Optional chaining _should_ be supported.

```suggestion
      event?.target?.value
```